### ISSUE TITLE
wsl.exe/bash return exit code 1 with no message to work off of.

### ISSUE THREAD

**Issue by IngwiePhoenix** [microsoft/WSL#5761]
# Environment

```none
Windows build number: 
Platform ServicePack Version      VersionString
-------- ----------- -------      -------------
 Win32NT             10.0.19041.0 Microsoft Windows NT 10.0.19041.0

Your Distribution version: 10.5
Whether the issue is on WSL 2 and/or WSL 1: WSL 2
```

# Steps to reproduce

After having run into several issues with my overall installation of Windows 10, I exported my Debian WSL image and re-imported it into `...home.../.wsl/Debian` using the `wsl` command itself. This yielded the following folder structure:

```
PS C:\Users\ingwi> tree /F .wsl
Auflistung der Ordnerpfade für Volume Windows HD
Volumeseriennummer : ACD3-4B55
C:\USERS\INGWI\.WSL
└───Debian
        ext4.vhdx
```

After that, I tried:
- `wsl -d Debian bash`
- `wsl bash`
- `bash`

Each of them exit with `%errorLevel%` of 1. And...that's it. No output, nothing.
```
C:\Users\ingwi>wsl -d Debian bash

C:\Users\ingwi>echo %errorLevel%
1

C:\Users\ingwi>wsl

C:\Users\ingwi>echo %errorLevel%
1

C:\Users\ingwi>bash

C:\Users\ingwi>echo %errorLevel%
1
``` 

**WSL logs**: https://aka.ms/AA9a5sp

#  Expected behavior

I expected to be able to just log back into my old WSL2 instance.

# Actual behavior

See above. TL;DR: Any command leads to an exit code 1.

**Comment by rintaun** [comment]
@IngwiePhoenix did you ever get any resolution on this?

**Comment by IngwiePhoenix** [comment]
> @IngwiePhoenix did you ever get any resolution on this?

Nein... I extracted the files from the tgz and grabbed what i wanted and planted it into a new WSL2 install. Quite a handful and annoying. Now I don't get my Windows' `PATH` in my WSL shell anymore - which isn't bad, but was a neat feature. So I probably have to hotchpotch that in again some other way.

**Comment by rintaun** [comment]
after a bit more searching, I was able to fix my problem. if you execute `wsl.exe --list`, I got something like the following:

```
PS C:\windows\system32> wsl -l
Windows Subsystem for Linux Distributions:
docker-desktop-data (Default)
ubuntu
docker-desktop
```

I think my problem was that `ubuntu` was not my default distribution.

I fixed that by executing `wsl.exe --set-default ubuntu`.

I was then able to run WSL normally, but it logged in as root rather than my user. I fixed that by following the instructions in https://github.com/microsoft/WSL/issues/4276 and https://github.com/microsoft/WSL/issues/3974, specifically:

executing `id -u <myuser>` inside wsl, I found uid `1000`, then executed:

```
Get-ItemProperty Registry::HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss\*\ DistributionName | Where-Object -Property DistributionName -eq ubuntu | Set-ItemProperty -Name DefaultUid -Value 1000
```
